.circleci/continue_config.yml

```diff
@@ -0,0 +1,597 @@
+
+  # This is effectively just a combination of the lint, unit_tests and e2e_tests jobs.
+  # It's used to check that the nightly docker image is working ok and before publishing a release.
+  build_kedro:
+    parameters:
+      python_version:
+        type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    steps:
+      - setup
+      - run:
+          name: Run linters
+          command: make lint
+      - run:
+          name: Run unit tests
+          command: make test
+      - run:
+          name: Run e2e tests
+          command: make e2e-tests
+
+  publish_kedro:
+    executor:
+      name: docker
+      python_version: "3.7"
+    steps:
+      - setup
+      - add_ssh_keys
+      - run:
+          name: Check Kedro version
+          command: |
+            KEDRO_VERSION=$(./tools/circleci/github_scripts/kedro_version.py ./kedro)
+
+            if ./tools/circleci/check-no-version-pypi.sh "${KEDRO_VERSION}"
+            then
+                echo "export KEDRO_VERSION=\"${KEDRO_VERSION}\"" >> $BASH_ENV
+            else
+                echo "Error: Kedro version ${KEDRO_VERSION} already exists on PyPI"
+                exit 1
+            fi
+      - run:
+          name: Tag and publish release on Github
+          command: ./tools/circleci/github_scripts/release.sh kedro-org kedro ${GITHUB_TAGGING_TOKEN} ${KEDRO_VERSION}
+      - run:
+          name: Publish to PyPI
+          command: |
+            make package
+            python -m pip install twine -U
+            python -m twine upload --repository-url ${TWINE_REPOSITORY_URL} dist/*
+      - run:
+          name: Trigger Read The Docs build
+          command: |
+            ./tools/circleci/rtd-build.sh ${RTD_TOKEN} stable
+            # give some time for GitHub release to propagate
+            # otherwise RTD fails to build a new tag
+            sleep 120
+            ./tools/circleci/rtd-build.sh ${RTD_TOKEN} ${KEDRO_VERSION}
+
+  # Trigger kedro-viz build to ensure tests in that project pass
+  viz_build:
+    docker:
+      - image: spotify/alpine # for bash and curl
+    steps:
+      - run:
+          name: Trigger kedro-viz build
+          command: |
+            curl --location --request POST \
+              --url https://circleci.com/api/v2/project/github/kedro-org/kedro-viz/pipeline \
+              --header "Circle-Token: $CIRCLE_VIZ_BUILD_TOKEN" \
+              --header 'content-type: application/json' \
+              --data '{"branch":"main"}'
+
+  all_circleci_checks_succeeded:
+    docker:
+      - image: circleci/python # any light-weight image
+    steps:
+      - run:
+          name: Success!
+          command: echo "All checks passed"
+
+workflows:
+  version: 2.1
+
+  build_docs:
+    when:
+      and:
+        - <<pipeline.parameters.build-docs>>
+        - not: <<pipeline.parameters.release_kedro>>
+    jobs:
+      - build_docs
```

Comment:
Very minor, but I find it a tiny bit confusing that there's a workflow called `build_docs` that calls `build_docs` + other jobs. I can't think of a good alternative name though 😅

Reply:
Yeah I thought about that but wanted to see if anyone else is bothered by it or I can be lazy lol. I'll rename one of them.